File: 2022-05/python/2_Add_Two_Numbers.py
class Solution:
    def addTwoNumbers(self, l1: Optional[ListNode], l2: Optional[ListNode]) -> Optional[ListNode]:
        res = cur = ListNode()
        carry = 0
        while l1 or l2 or carry:
            if l1:
                carry += l1.val
                l1 = l1.next
            if l2:
                carry += l2.val
                l2 = l2.next
            carry, val = divmod(carry, 10)
            cur.next = ListNode(val)
            cur = cur.next
        return res.next

File: 2022-05/typescript/2_Add_Two_Numbers.ts
function addTwoNumbers(l1: null | ListNode, l2: null | ListNode): null | ListNode {
    const res = new ListNode()
    let cur = res
    let carry = 0
    while (l1 || l2 || carry) {
        if (l1) {
            carry += l1.val
            l1 = l1.next
        }
        if (l2) {
            carry += l2.val
            l2 = l2.next
        }
        cur.next = new ListNode(carry % 10)
        cur = cur.next
        carry = Math.trunc(carry / 10)
    }
    return res.next
}